src/main/features/ccj/form/models/repaymentPlan.ts

```diff
@@ -0,0 +1,76 @@
+import { LocalDate } from 'forms/models/localDate'
+import { PaymentSchedule } from 'ccj/form/models/paymentSchedule'
+import { IsDefined, IsIn, IsPositive, ValidateNested } from 'class-validator'
+import { isValidYearFormat } from 'forms/validation/validators/isValidYearFormat'
+import { IsValidLocalDate } from 'forms/validation/validators/isValidLocalDate'
+import { IsFutureDate } from 'app/forms/validation/validators/dateFutureConstraint'
+import { IsLessThan } from 'forms/validation/validators/isLessThan'
+
+export class ValidationErrors {
+  static readonly FIRST_PAYMENT_AMOUNT_INVALID: string = 'Enter a valid amount of first payment'
+  static readonly INSTALMENTS_AMOUNT_INVALID: string = 'Enter a valid amount for equal instalments'
+  static readonly FUTURE_DATE: string = 'Enter a first payment date in the future'
+  static readonly INVALID_DATE: string = 'Enter a valid date of first payment'
+  static readonly SELECT_PAYMENT_SCHEDULE: string = 'Select how often they should pay'
+}
+
+export class RepaymentPlan {
+
+  remainingAmount?: number
+
+  @IsPositive({ message: ValidationErrors.FIRST_PAYMENT_AMOUNT_INVALID })
+  @IsLessThan('remainingAmount', { message: ValidationErrors.FIRST_PAYMENT_AMOUNT_INVALID })
+  firstPayment?: number
+
+  @IsPositive({ message: ValidationErrors.INSTALMENTS_AMOUNT_INVALID })
+  @IsLessThan('remainingAmount', { message: ValidationErrors.INSTALMENTS_AMOUNT_INVALID })
+  installmentAmount?: number
+
+  @ValidateNested()
+  @IsDefined({ message: ValidationErrors.INVALID_DATE })
+  @IsValidLocalDate({ message: ValidationErrors.INVALID_DATE })
+  @isValidYearFormat(4, { message: ValidationErrors.INVALID_DATE })
+  @IsFutureDate({ message: ValidationErrors.FUTURE_DATE })
+  firstPaymentDate?: LocalDate
+
+  @IsIn(PaymentSchedule.all(), { message: ValidationErrors.SELECT_PAYMENT_SCHEDULE })
+  paymentSchedule?: PaymentSchedule
+
+  constructor (remainingAmount?: number, firstPayment?: number, installmentAmount?: number, firstPaymentDate?: LocalDate, paymentSchedule?: PaymentSchedule) {
+    this.remainingAmount = remainingAmount
+    this.firstPayment = firstPayment
+    this.installmentAmount = installmentAmount
+    this.firstPaymentDate = firstPaymentDate
+    this.paymentSchedule = paymentSchedule
+  }
+
+  static fromObject (value?: any): RepaymentPlan {
+    if (value) {
+      const remainingAmount = value.remainingAmount ? parseFloat(value.remainingAmount) : undefined
```

Comment:
what would you say about some util static method to parseNumbers that would do what's done here?

Reply:
maybe